projects/tsq-gallery/src/lib/components/tsq-gallery-viewer/tsq-gallery-viewer.component.ts

```diff
@@ -60,192 +53,152 @@ export class TSqGalleryViewerComponent {
-    return this.imageZoom > 1.0;
-  }
+  goBack(addDelay = false) {
+    if (this.canGoBack && !this.navigationDelay) {
+      this.setCurrentFile(this.selectedFileIndex - 1);
 
-  get canZoomIn(): boolean {
-    return this.imageZoom < 1.8;
+      if (addDelay) {
+        this.navigationDelay = true;
+        setTimeout(() => this.navigationDelay = false, 150);
+      }
+    }
   }
 
-  get canZoomOut(): boolean {
-    return this.imageZoom > 0.5;
-  }
+  closeBackdrop($event: TouchEvent | MouseEvent) {
+    $event.stopPropagation();
 
-  get canGoFoward(): boolean {
-    return this.selectedFileIndex < this.files.length - 1 && !this.navigationDelay;
+    if (this.backdropClickClose && !this.isMoving && !this.showLoading) {
+      if (!this.keepOpen) {
+        this.close();
+      } else if (this.expanded) {
+        this.expanded = false;
+      }
+    }
   }
 
-  get canGoBack(): boolean {
-    return this.selectedFileIndex > 0 && !this.navigationDelay;
+  turnLeft() {
+    this.imageRotation -= 90;
   }
 
-  resetPosition(resetRotation: boolean = true) {
-    this.imageZoom = 1;
-    this.initialX = 0;
-    this.initialY = 0;
-    this.initialLeft = 0;
-    this.initialTop = 0;
-    this.positionLeft = 0;
-    this.positionTop = 0;
-
-    if (resetRotation) {
-      this.imageRotation = 0;
-    }
+  turnRight() {
+    this.imageRotation += 90;
   }
 
-  open(index?: number) {
-    this.isOpen = true;
-    this.selectedFileIndex = index || 0;
-    this.renderer.setStyle(document.body, 'overflow', 'hidden');
-    setTimeout(() => {
-      if (!!this.backdropRef) {
-        this.backdropRef.nativeElement.focus();
-      }
-
-      this.resetPosition();
-    }, 10);
-
-    const throttleFunc = throttle(this.onWindowScroll, 1);
-    this.scrollListener = this.renderer.listen('document', 'wheel', throttleFunc);
+  downloadFile() {
+    const aTag = document.createElement('a');
+    aTag.setAttribute('download', this.selectedFileToDisplay.name);
+    aTag.setAttribute('href', this.selectedFileToDisplay.downloadUrl);
+    aTag.setAttribute('type', 'hidden');
+    aTag.setAttribute('target', '_blank');
 
-    this.renderer.setStyle(document.body, 'overscroll-behavior-x', 'none');
+    document.body.appendChild(aTag);
+    aTag.click();
+    aTag.remove();
   }
 
-  close() {
-    this.resetPosition();
-
-    this.isOpen = false;
-    this.selectedFileIndex = undefined;
-    this.renderer.removeStyle(document.body, 'overflow');
-
-    if (!!this.scrollListener) {
-      this.scrollListener();
+  zoomIn() {
+    if (this.canZoomIn) {
+      this.setImageZoom(0.1);
     }
+  }
 
-    this.renderer.removeClass(document.body, 'overscroll-behavior-x');
+  zoomOut() {
+    if (this.canZoomOut) {
+      this.setImageZoom(-0.1);
+      this.adjustHorizontalPositionOnZoom();
+    }
   }
 
   onWindowScroll = ($event: WheelEvent) => {
+    $event.preventDefault();
+
     if ($event.ctrlKey) {
+      // tslint:disable-next-line: no-string-literal
```

Comment:
I don't see this as a good practice. If it's needed is better to modify this rule directly on tslint config file.

Reply:
I was thinking on doing something like that, but I came to the conclusion that I would end up just moving literally everything to that class which would result in the same problem there. I am thinking if there as a way to do it still